File: api/src/cache/TrendsFileCache.ts
import RedisCache from './RedisCache'

const TrendsFileCache_TTL = 60 * 60 * 3 // 3 hours

export const TrendsFileCache = {
  async get(key: string): Promise<string | null> {
    const value = await RedisCache.get(composeKey(key))
    return value
  },

  async set(key: string, value: string): Promise<void> {
    await RedisCache.set(composeKey(key), value, 'EX', TrendsFileCache_TTL)
  }
}

const composeKey = (storageKey: string): string => {
  return `trends-file:${storageKey}`
}

File: api/src/main/java/com/danmoop/datanow/Cache/TrendsFileCache.java
package com.danmoop.datanow.Cache;

import org.springframework.stereotype.Service;

import java.time.Duration;
import java.util.Optional;

@Service
public class TrendsFileCache {

  private final Duration TTL = Duration.ofHours(1);
  private final RedisCache redis;

  public TrendsFileCache(RedisCache redis) {
    this.redis = redis;
  }

  public void set(String storageKey, String value) {
    redis.set(composeKey(storageKey), value, TTL);
  }

  public Optional<String> get(String storageKey) {
    return redis.get(composeKey(storageKey));
  }

  public void delete(String storageKey) {
    redis.delete(composeKey(storageKey));
  }

  private String composeKey(String storageKey) {
    return "trends-file:" + storageKey;
  }
}
